superset/views/datasource.py

```diff
@@ -53,7 +53,11 @@ def save(self) -> FlaskResponse:
         )
         orm_datasource.database_id = database_id
 
-        if "owners" in datasource_dict and orm_datasource.owner_class is not None:
+        if (
+            app.config["OLD_API_CHECK_DATASET_OWNERSHIP"]
```

Comment:
nit: I think we want to keep the part of this that sets `datasource_dict["owners"]` and only disable the `check_ownership` function